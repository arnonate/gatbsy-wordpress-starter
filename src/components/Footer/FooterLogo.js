import React from "react"

const FooterLogo = () => (
  <svg
    width="2700"
    height="1200"
    viewBox="0 0 2700 1200"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1426.48 363.38C1432.24 363.38 1435.47 367.86 1433.66 373.33L1354.52 612.13C1352.71 617.6 1346.51 622.08 1340.75 622.08H1310.71C1304.95 622.08 1298.74 617.61 1296.91 612.14L1239.61 440.58C1237.78 435.11 1234.77 435.11 1232.92 440.56L1174.62 612.15C1172.77 617.61 1166.53 622.07 1160.77 622.07H1130.36C1124.6 622.07 1118.4 617.59 1116.59 612.12L1037.44 373.32C1035.63 367.85 1038.86 363.37 1044.62 363.37H1073.56C1079.32 363.37 1085.5 367.85 1087.29 373.33L1145.08 550.02C1146.87 555.5 1149.86 555.51 1151.71 550.06L1211.85 373.29C1213.71 367.83 1219.94 363.37 1225.7 363.37H1249.09C1254.85 363.37 1261.05 367.85 1262.86 373.32L1321.69 551.14C1323.5 556.61 1326.48 556.62 1328.31 551.15L1387.81 373.31C1389.64 367.85 1395.85 363.37 1401.61 363.37H1426.48V363.38Z"
      fill="white"
    />
    <path
      d="M1621.21 571.83C1618.92 566.54 1612.34 562.21 1606.58 562.21H1498.19C1492.43 562.21 1485.86 566.55 1483.61 571.85L1466.32 612.44C1464.06 617.74 1457.5 622.08 1451.74 622.08H1423.17C1417.41 622.08 1414.63 617.78 1416.99 612.53L1524.8 372.95C1527.16 367.69 1533.81 363.4 1539.58 363.4H1565.93C1571.69 363.4 1578.35 367.7 1580.72 372.95L1688.88 612.54C1691.25 617.79 1688.48 622.09 1682.71 622.09H1653.41C1647.65 622.09 1641.06 617.76 1638.78 612.47L1621.21 571.83ZM1590.69 524.52C1596.45 524.52 1599.3 520.19 1597.02 514.9L1556.54 421.05C1554.26 415.76 1550.53 415.76 1548.27 421.06L1508.1 514.88C1505.83 520.18 1508.69 524.51 1514.45 524.51H1590.69V524.52Z"
      fill="white"
    />
    <path
      d="M1940.95 363.38C1946.71 363.38 1951.43 368.09 1951.43 373.86V611.6C1951.43 617.36 1946.71 622.08 1940.95 622.08H1922.36C1916.6 622.08 1908.9 618.42 1905.27 613.96L1775.85 455.04C1772.21 450.57 1769.23 451.63 1769.23 457.39V611.6C1769.23 617.36 1764.52 622.08 1758.75 622.08H1732.03C1726.27 622.08 1721.55 617.36 1721.55 611.6V373.86C1721.55 368.1 1726.27 363.38 1732.03 363.38H1750.62C1756.38 363.38 1764.08 367.04 1767.71 371.5L1897.13 530.42C1900.77 534.89 1903.75 533.83 1903.75 528.07V373.86C1903.75 368.1 1908.46 363.38 1914.23 363.38H1940.95Z"
      fill="white"
    />
    <path
      d="M2222.22 363.38C2227.98 363.38 2232.7 368.09 2232.7 373.86V611.6C2232.7 617.36 2227.99 622.08 2222.22 622.08H2203.63C2197.87 622.08 2190.17 618.42 2186.54 613.96L2057.12 455.04C2053.48 450.57 2050.5 451.63 2050.5 457.39V611.6C2050.5 617.36 2045.78 622.08 2040.02 622.08H2013.3C2007.54 622.08 2002.82 617.36 2002.82 611.6V373.86C2002.82 368.1 2007.54 363.38 2013.3 363.38H2031.89C2037.65 363.38 2045.35 367.04 2048.98 371.5L2178.4 530.42C2182.04 534.89 2185.02 533.83 2185.02 528.07V373.86C2185.02 368.1 2189.74 363.38 2195.5 363.38H2222.22Z"
      fill="white"
    />
    <path
      d="M2474.12 571.83C2471.83 566.54 2465.25 562.21 2459.49 562.21H2351.1C2345.34 562.21 2338.77 566.55 2336.52 571.85L2319.23 612.44C2316.97 617.74 2310.41 622.08 2304.65 622.08H2276.08C2270.32 622.08 2267.54 617.78 2269.9 612.53L2377.71 372.95C2380.07 367.69 2386.72 363.4 2392.49 363.4H2418.84C2424.6 363.4 2431.26 367.7 2433.63 372.95L2541.79 612.54C2544.16 617.79 2541.39 622.09 2535.62 622.09H2506.32C2500.56 622.09 2493.97 617.76 2491.69 612.47L2474.12 571.83ZM2443.59 524.52C2449.35 524.52 2452.2 520.19 2449.92 514.9L2409.44 421.05C2407.16 415.76 2403.43 415.76 2401.17 421.06L2361 514.88C2358.73 520.18 2361.59 524.51 2367.35 524.51H2443.59V524.52Z"
      fill="white"
    />
    <path
      d="M1094.92 896.22C1089.12 892.7 1084.66 887.7 1081.54 881.22C1078.42 874.74 1076.86 867.11 1076.86 858.31C1076.86 849.43 1078.4 841.74 1081.48 835.22C1084.56 828.7 1089 823.7 1094.8 820.22C1100.6 816.74 1107.46 815 1115.37 815C1123.29 815 1130.14 816.74 1135.94 820.22C1141.74 823.7 1146.18 828.7 1149.26 835.22C1152.34 841.74 1153.88 849.4 1153.88 858.19C1153.88 866.99 1152.32 874.65 1149.2 881.16C1146.08 887.68 1141.62 892.7 1135.82 896.22C1130.02 899.74 1123.2 901.5 1115.37 901.5C1107.54 901.5 1100.72 899.74 1094.92 896.22ZM1136.31 884.04C1141.31 878 1143.81 869.39 1143.81 858.19C1143.81 846.99 1141.33 838.4 1136.37 832.4C1131.41 826.4 1124.41 823.4 1115.38 823.4C1106.42 823.4 1099.44 826.42 1094.45 832.46C1089.45 838.5 1086.95 847.08 1086.95 858.19C1086.95 869.31 1089.45 877.91 1094.45 883.98C1099.45 890.06 1106.43 893.1 1115.38 893.1C1124.33 893.1 1131.31 890.08 1136.31 884.04Z"
      fill="white"
    />
    <path
      d="M1230.18 815.96H1239.42V900.54H1231.86L1180.88 832.52V900.54H1171.64V815.96H1179.2L1230.18 883.98V815.96Z"
      fill="white"
    />
    <path
      d="M1260.29 900.54V815.96H1270.13V892.14H1312.48V900.54H1260.29V900.54Z"
      fill="white"
    />
    <path d="M1325.07 900.54V815.96H1334.91V900.54H1325.07Z" fill="white" />
    <path
      d="M1414.33 815.96H1423.57V900.54H1416.01L1365.03 832.52V900.54H1355.79V815.96H1363.35L1414.33 883.98V815.96Z"
      fill="white"
    />
    <path
      d="M1444.44 900.54V815.96H1497.94V824.12H1454.15V853.51H1495.54V861.67H1454.15V892.38H1497.94V900.54H1444.44V900.54Z"
      fill="white"
    />
    <path
      d="M1540.41 890.82L1544.25 883.26C1548.41 886.78 1552.67 889.3 1557.03 890.82C1561.39 892.34 1566.45 893.1 1572.21 893.1C1579.01 893.1 1584.25 891.78 1587.93 889.14C1591.61 886.5 1593.45 882.78 1593.45 877.98C1593.45 875.18 1592.57 872.9 1590.81 871.14C1589.05 869.38 1586.73 867.98 1583.85 866.94C1580.97 865.9 1577.09 864.82 1572.21 863.7C1565.65 862.18 1560.27 860.58 1556.08 858.9C1551.88 857.22 1548.48 854.82 1545.88 851.7C1543.28 848.58 1541.98 844.5 1541.98 839.46C1541.98 834.66 1543.24 830.4 1545.76 826.68C1548.28 822.96 1551.88 820.08 1556.56 818.04C1561.24 816 1566.66 814.98 1572.81 814.98C1578.65 814.98 1584.11 815.9 1589.19 817.74C1594.27 819.58 1598.57 822.22 1602.09 825.66L1598.37 833.22C1594.37 829.78 1590.31 827.28 1586.19 825.72C1582.07 824.16 1577.61 823.38 1572.81 823.38C1566.25 823.38 1561.11 824.76 1557.39 827.52C1553.67 830.28 1551.81 834.14 1551.81 839.1C1551.81 843.34 1553.45 846.54 1556.73 848.7C1560.01 850.86 1564.97 852.7 1571.61 854.22C1578.89 855.98 1584.61 857.6 1588.77 859.08C1592.93 860.56 1596.39 862.78 1599.15 865.74C1601.91 868.7 1603.29 872.66 1603.29 877.62C1603.29 882.42 1602.03 886.62 1599.51 890.22C1596.99 893.82 1593.37 896.6 1588.65 898.56C1583.93 900.52 1578.41 901.5 1572.09 901.5C1558.88 901.5 1548.32 897.94 1540.41 890.82Z"
      fill="white"
    />
    <path
      d="M1619.22 815.96H1652.69C1661.57 815.96 1668.47 818.12 1673.38 822.44C1678.3 826.76 1680.76 832.88 1680.76 840.79C1680.76 848.7 1678.28 854.87 1673.32 859.26C1668.36 863.66 1661.48 865.86 1652.69 865.86H1629.06V900.53H1619.22V815.96ZM1651.73 857.71C1664.61 857.71 1671.04 852.07 1671.04 840.79C1671.04 829.67 1664.6 824.12 1651.73 824.12H1629.06V857.71H1651.73V857.71Z"
      fill="white"
    />
    <path
      d="M1710.69 896.22C1704.89 892.7 1700.43 887.7 1697.31 881.22C1694.19 874.74 1692.63 867.11 1692.63 858.31C1692.63 849.43 1694.17 841.74 1697.25 835.22C1700.33 828.7 1704.77 823.7 1710.57 820.22C1716.37 816.74 1723.23 815 1731.14 815C1739.05 815 1745.91 816.74 1751.71 820.22C1757.51 823.7 1761.95 828.7 1765.03 835.22C1768.11 841.74 1769.65 849.4 1769.65 858.19C1769.65 866.99 1768.09 874.65 1764.97 881.16C1761.85 887.68 1757.39 892.7 1751.59 896.22C1745.79 899.74 1738.97 901.5 1731.14 901.5C1723.31 901.5 1716.49 899.74 1710.69 896.22ZM1752.08 884.04C1757.08 878 1759.58 869.39 1759.58 858.19C1759.58 846.99 1757.1 838.4 1752.14 832.4C1747.18 826.4 1740.18 823.4 1731.15 823.4C1722.19 823.4 1715.21 826.42 1710.22 832.46C1705.22 838.5 1702.72 847.08 1702.72 858.19C1702.72 869.31 1705.22 877.91 1710.22 883.98C1715.22 890.06 1722.2 893.1 1731.15 893.1C1740.1 893.1 1747.08 890.08 1752.08 884.04Z"
      fill="white"
    />
    <path
      d="M1853.99 900.54H1843.19L1828.79 873.43C1827.03 870.15 1824.97 867.89 1822.61 866.65C1820.25 865.41 1817.03 864.79 1812.95 864.79H1797.12V900.54H1787.4V815.96H1822.19C1831.39 815.96 1838.4 818 1843.24 822.08C1848.08 826.16 1850.5 832.16 1850.5 840.08C1850.5 846.64 1848.64 851.94 1844.92 855.98C1841.2 860.02 1835.9 862.56 1829.02 863.6C1831.18 864.24 1833.08 865.32 1834.72 866.84C1836.36 868.36 1837.86 870.4 1839.22 872.96L1853.99 900.54ZM1821.36 856.63C1827.92 856.63 1832.82 855.27 1836.06 852.55C1839.3 849.83 1840.92 845.71 1840.92 840.19C1840.92 834.75 1839.32 830.71 1836.12 828.07C1832.92 825.43 1828 824.11 1821.36 824.11H1797.13V856.62H1821.36V856.63Z"
      fill="white"
    />
    <path
      d="M1887.7 900.54V824.36H1858.31V815.96H1927.17V824.36H1897.66V900.54H1887.7Z"
      fill="white"
    />
    <path
      d="M1931.37 890.82L1935.21 883.26C1939.37 886.78 1943.63 889.3 1947.99 890.82C1952.35 892.34 1957.41 893.1 1963.17 893.1C1969.97 893.1 1975.21 891.78 1978.89 889.14C1982.57 886.5 1984.41 882.78 1984.41 877.98C1984.41 875.18 1983.53 872.9 1981.77 871.14C1980.01 869.38 1977.69 867.98 1974.81 866.94C1971.93 865.9 1968.05 864.82 1963.17 863.7C1956.61 862.18 1951.23 860.58 1947.04 858.9C1942.84 857.22 1939.44 854.82 1936.84 851.7C1934.24 848.58 1932.94 844.5 1932.94 839.46C1932.94 834.66 1934.2 830.4 1936.72 826.68C1939.24 822.96 1942.84 820.08 1947.52 818.04C1952.2 816 1957.62 814.98 1963.77 814.98C1969.61 814.98 1975.07 815.9 1980.15 817.74C1985.23 819.58 1989.53 822.22 1993.05 825.66L1989.33 833.22C1985.33 829.78 1981.27 827.28 1977.15 825.72C1973.03 824.16 1968.57 823.38 1963.77 823.38C1957.21 823.38 1952.07 824.76 1948.35 827.52C1944.63 830.28 1942.77 834.14 1942.77 839.1C1942.77 843.34 1944.41 846.54 1947.69 848.7C1950.97 850.86 1955.93 852.7 1962.57 854.22C1969.85 855.98 1975.57 857.6 1979.73 859.08C1983.89 860.56 1987.35 862.78 1990.11 865.74C1992.87 868.7 1994.25 872.66 1994.25 877.62C1994.25 882.42 1992.99 886.62 1990.47 890.22C1987.95 893.82 1984.33 896.6 1979.61 898.56C1974.89 900.52 1969.37 901.5 1963.05 901.5C1949.84 901.5 1939.29 897.94 1931.37 890.82Z"
      fill="white"
    />
    <path
      d="M2100.46 864.31C2103.14 867.83 2104.48 872.23 2104.48 877.51C2104.48 884.95 2102.04 890.65 2097.16 894.6C2092.28 898.56 2085.36 900.54 2076.41 900.54H2041.14V815.96H2075.09C2083.81 815.96 2090.57 817.88 2095.36 821.72C2100.16 825.56 2102.56 831 2102.56 838.04C2102.56 842.6 2101.38 846.52 2099.02 849.8C2096.66 853.08 2093.36 855.52 2089.12 857.12C2094 858.39 2097.78 860.79 2100.46 864.31ZM2050.85 853.51H2073.64C2079.96 853.51 2084.74 852.23 2087.98 849.67C2091.22 847.11 2092.84 843.39 2092.84 838.51C2092.84 828.91 2086.44 824.11 2073.65 824.11H2050.86V853.51H2050.85ZM2090.14 888.72C2093.3 886.28 2094.88 882.46 2094.88 877.26C2094.88 866.86 2088.4 861.67 2075.45 861.67H2050.86V892.38H2075.45C2082.08 892.38 2086.98 891.16 2090.14 888.72Z"
      fill="white"
    />
    <path
      d="M2121.99 900.54V815.96H2175.49V824.12H2131.7V853.51H2173.09V861.67H2131.7V892.38H2175.49V900.54H2121.99V900.54Z"
      fill="white"
    />
    <path
      d="M2210.76 900.54V824.36H2181.37V815.96H2250.23V824.36H2220.72V900.54H2210.76Z"
      fill="white"
    />
    <path
      d="M2282.5 900.54V824.36H2253.11V815.96H2321.97V824.36H2292.46V900.54H2282.5Z"
      fill="white"
    />
    <path d="M2333.84 900.54V815.96H2343.68V900.54H2333.84Z" fill="white" />
    <path
      d="M2423.1 815.96H2432.34V900.54H2424.78L2373.79 832.52V900.54H2364.55V815.96H2372.11L2423.09 883.98V815.96H2423.1Z"
      fill="white"
    />
    <path
      d="M2520.15 856.51V895.62C2516.79 897.38 2512.41 898.8 2507.01 899.88C2501.61 900.96 2496.15 901.5 2490.63 901.5C2482.23 901.5 2474.99 899.78 2468.92 896.34C2462.84 892.9 2458.18 887.94 2454.94 881.46C2451.7 874.98 2450.08 867.23 2450.08 858.19C2450.08 849.23 2451.68 841.52 2454.88 835.04C2458.08 828.56 2462.68 823.6 2468.68 820.16C2474.68 816.72 2481.76 815 2489.91 815C2495.83 815 2501.27 815.9 2506.23 817.7C2511.19 819.5 2515.43 822.16 2518.95 825.68L2515.23 833.24C2511.15 829.72 2507.13 827.2 2503.17 825.68C2499.21 824.16 2494.79 823.4 2489.91 823.4C2480.39 823.4 2473.07 826.4 2467.96 832.4C2462.84 838.4 2460.28 847 2460.28 858.19C2460.28 869.63 2462.88 878.33 2468.08 884.28C2473.28 890.24 2480.87 893.22 2490.87 893.22C2498.15 893.22 2504.91 892.14 2511.14 889.98V864.19H2490.15V856.51H2520.15V856.51Z"
      fill="white"
    />
    <path
      d="M752.87 850.52C726.23 909.16 656.45 935.34 597.8 908.7L173.04 715.72C114.4 689.08 88.2201 619.3 114.86 560.65L307.84 135.88C334.48 77.24 404.26 51.0599 462.91 77.6999L887.68 270.68C946.32 297.32 972.5 367.1 945.86 425.75L752.87 850.52Z"
      fill="#FBD448"
    />
    <path
      d="M941.77 611.85C962.11 672.97 928.75 739.61 867.63 759.95L424.95 907.27C363.83 927.61 297.19 894.25 276.85 833.13L129.53 390.45C109.19 329.33 142.55 262.69 203.67 242.35L646.35 95.0299C707.47 74.6899 774.11 108.05 794.45 169.17L941.77 611.85Z"
      fill="#7507CA"
    />
    <path
      d="M734.07 371.98C743.41 371.98 748.56 379.2 745.5 388.03L661.17 631.42C658.11 640.25 647.96 647.47 638.62 647.47H614.46C605.12 647.47 594.93 640.26 591.83 631.45L536.25 473.64C533.15 464.83 528.03 464.82 524.89 473.61L468.49 631.47C465.35 640.27 455.13 647.47 445.79 647.47H421.6C412.26 647.47 402.12 640.24 399.08 631.41L315.19 388.04C312.14 379.21 317.3 371.98 326.64 371.98H349.84C359.18 371.98 369.26 379.23 372.24 388.08L429.73 559.14C432.71 568 437.67 568.03 440.76 559.21L500.75 388.01C503.84 379.19 514.01 371.98 523.36 371.98H540.26C549.6 371.98 559.72 379.21 562.74 388.06L621.53 560.21C624.55 569.05 629.51 569.06 632.55 560.22L691.76 388.05C694.8 379.21 704.93 371.99 714.27 371.99H734.07V371.98Z"
      fill="white"
    />
  </svg>
)

export default FooterLogo
