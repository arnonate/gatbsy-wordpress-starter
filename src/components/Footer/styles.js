import styled from "styled-components"

import { Tokens } from "../Theme/Tokens"

export const Download = styled.div`
  padding: 0;
  border-top: 1px solid ${Tokens.color.border};

  @media (min-width: ${Tokens.screen.medium}) {
    padding: calc(${Tokens.rhythm} * 2);
  }
`

export const Footer = styled.footer`
  position: relative;
  overflow: hidden;
  padding: calc(${Tokens.rhythm} * 2) 0;
  background: rgb(49, 49, 52);
  background: radial-gradient(
    circle,
    rgba(49, 49, 52, 1) 0%,
    rgba(31, 31, 33, 1) 100%
  );

  p {
    font-size: ${Tokens.font.size.micro};
    color: ${Tokens.color.white};
    opacity: 0.4;
  }

  a {
    color: ${Tokens.color.white};
  }
`

export const LogoSection = styled.div`
  width: 100%;
  text-align: center;

  @media (min-width: ${Tokens.screen.medium}) {
    width: 20%;
    text-align: left;
  }
`

export const LinkSection = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;

  @media (min-width: ${Tokens.screen.medium}) {
    width: 30%;
  }
`

export const CopySection = styled.div`
  width: 95%;
  margin: auto;

  @media (min-width: ${Tokens.screen.medium}) {
    width: 35%;
    margin: 0;
  }
`

export const Logo = styled.div`
  width: 100%;
  margin: auto;

  @media (min-width: ${Tokens.screen.medium}) {
    margin: 0;
  }

  svg {
    width: 100%;
    height: auto;
  }
`

export const LinkGroup = styled.div`
  width: 50%;
  margin: auto;
  padding-bottom: calc(${Tokens.rhythm} * 2);

  @media (min-width: ${Tokens.screen.medium}) {
    margin: 0;
    padding-bottom: 0;
  }

  a {
    display: block;
    text-decoration: none;
    font-weight: ${Tokens.font.weight.bold};
    text-align: center;

    @media (min-width: ${Tokens.screen.medium}) {
      text-align: left;
    }
  }

  a + a {
    padding-top: ${Tokens.rhythm};
  }
`

export const Gaming = styled.div`
  width: 150px;
`

export const Background = styled.div`
  position: absolute;
  top: 0;
  left: 50%;
  min-width: 100%;
  min-height: 100%;
  overflow: visible;
  transform: translateX(-50%);
  /* z-index: ${Tokens.z.behind}; */

  background: radial-gradient(
    circle,
    rgba(255, 255, 255, 0.1026611328125) 0%,
    rgba(31, 31, 33, 1) 100%
  );
`
