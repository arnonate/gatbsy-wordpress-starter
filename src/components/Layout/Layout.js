import React, { useState, useEffect } from "react"
import { ParallaxProvider } from "react-scroll-parallax"
import { navigate } from "gatsby"
import { createGlobalStyle } from "styled-components"

import { Context } from "../Context"
import "../Theme/reset.css"
import {
  Footer,
  Header,
  AppModal,
  LanguageModal,
  MenuModal,
} from "../../components"

const Layout = ({ children }) => {
  const defaultLang =
    (typeof window !== "undefined" &&
      window.localStorage.getItem("wannaLang")) ||
    (typeof window !== "undefined" && window.navigator.language.split("-")[0])

  const [language, toggleLanguage] = useState(defaultLang ? defaultLang : "en")
  const [appModalIsVisible, toggleAppModalIsVisible] = useState(false)
  const [languageModalIsVisible, toggleLanguageModalIsVisible] = useState(false)
  const [menuModalIsVisible, toggleMenuModalIsVisible] = useState(false)

  useEffect(() => {
    const langList = ["pt"]

    const path =
      typeof window !== "undefined" &&
      window.location.pathname
        .split("/")
        .filter(item => item.length && !langList.includes(item))

    const hash =
      typeof window !== "undefined" && window.location.hash
        ? window.location.hash
        : ""

    if (language !== "en" && !path.includes(language)) {
      const uri = `/${language ? language + "/" : ""}${
        path.length > 0 ? path.join("/") + "/" : ""
      }${hash}`
      navigate(uri)
    }
    if (language === "en") {
      const uri = `/${path.length > 0 ? path.join("/") + "/" : ""}${hash}`
      navigate(uri)
    }
  }, [language])

  const context = {
    language,
    toggleAppModalIsVisible,
    appModalIsVisible,
    toggleLanguage,
    languageModalIsVisible,
    toggleLanguageModalIsVisible,
    menuModalIsVisible,
    toggleMenuModalIsVisible,
  }

  const updateLanguage = lang => {
    toggleLanguage(lang)
    toggleLanguageModalIsVisible(false)
    toggleMenuModalIsVisible(false)

    if (typeof window !== "undefined") {
      window.localStorage.setItem("wannaLang", lang)
    }
  }

  return (
    <Context.Provider value={context}>
      <GlobalStyle
        modalIsVisible={languageModalIsVisible || menuModalIsVisible}
      />

      <Header />

      <ParallaxProvider>
        <main>{children}</main>
      </ParallaxProvider>

      <Footer />

      <AppModal isVisible={appModalIsVisible} />

      <LanguageModal
        isVisible={languageModalIsVisible}
        onUpdate={updateLanguage}
        onClose={toggleLanguageModalIsVisible}
        currentLanguage={language}
      />

      <MenuModal
        isVisible={menuModalIsVisible}
        onLanguageUpdate={updateLanguage}
      />
    </Context.Provider>
  )
}

const GlobalStyle = createGlobalStyle`
  body {
    overflow: ${props => (props.modalIsVisible ? "hidden" : "auto")};
  }
`

export default Layout
