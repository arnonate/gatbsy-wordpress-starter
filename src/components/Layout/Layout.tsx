import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { FluidObject } from "gatsby-image"

import { Footer, Header } from "../../components"
import "../../styles/reset.css"
import "../../styles/global.css"

export type DataProps = {
  logo: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
  site: {
    siteMetadata: {
      title: string
      author: string
    }
  }
}

type ComponentProps = {
  children: React.ReactNode
  data?: Readonly<DataProps>
}

export const LayoutComponent = ({
  data,
  children,
}: Readonly<ComponentProps>): JSX.Element => (
  <>
    <Header data={data} />
    <main>
      <div className="girdle">{children}</div>
    </main>
    <Footer data={data} />
  </>
)

const Layout = ({ children }: ComponentProps): JSX.Element => {
  const data = useStaticQuery<DataProps>(graphql`
    query {
      logo: file(relativePath: { eq: "logo.png" }) {
        childImageSharp {
          fluid(maxWidth: 400) {
            ...GatsbyImageSharpFluid_withWebp_noBase64
          }
        }
      }
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `)

  return <LayoutComponent data={data}>{children}</LayoutComponent>
}

export default Layout
