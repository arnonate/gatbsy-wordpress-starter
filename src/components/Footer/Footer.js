import React from "react"
import { Link, graphql, useStaticQuery } from "gatsby"
import { Context } from "../Context"
import Logo from "./FooterLogo"

import { AppDownload, SocialLinks, Theme } from "../index"
import * as Styles from "./styles.js"

const Footer = () => {
  const data = useStaticQuery(graphql`
    query {
      bg: file(relativePath: { eq: "footer-bg.jpg" }) {
        childImageSharp {
          fluid(maxWidth: 1800) {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
      wp: wpPage(slug: { eq: "footer" }) {
        footerFields {
          footerFields {
            languageValue {
              name
              slug
            }
            linkGroup2 {
              label
              link
            }
            linkGroup1 {
              label
              link
            }
            copy
          }
        }
      }
    }
  `)

  return (
    <Context.Consumer>
      {({ language }) => {
        const fields = data.wp.footerFields.footerFields.find(row =>
          row.languageValue.slug.includes(language)
        )

        return (
          <>
            <Styles.Download id="scrollTo--downloads">
              <AppDownload />
            </Styles.Download>
            <Styles.Footer>
              <Theme.Wrap size="wide" flex="row">
                <Styles.LogoSection>
                  <Styles.Logo>
                    <Link to={language === "pt" ? "/pt/" : "/"}>
                      {/* <Img
                        fluid={data.logo.childImageSharp.fluid}
                        alt="Wanna"
                      /> */}
                      <Logo />
                    </Link>
                  </Styles.Logo>

                  <SocialLinks fill="white" />
                </Styles.LogoSection>

                <Styles.LinkSection>
                  <Styles.LinkGroup>
                    {fields.linkGroup1.map(link => (
                      <Link key={link.label} to={link.link}>
                        {link.label}
                      </Link>
                    ))}
                  </Styles.LinkGroup>
                  <Styles.LinkGroup>
                    {fields.linkGroup2.map(link => (
                      <Link key={link.label} to={link.link}>
                        {link.label}
                      </Link>
                    ))}
                  </Styles.LinkGroup>
                </Styles.LinkSection>

                <Styles.CopySection>
                  <Styles.Gaming>
                    <iframe
                      src="https://licensing.gaming-curacao.com/validator/?lh=9f83b305d6344e5c95267ec34f2d8789&template=tseal"
                      width={150}
                      height={50}
                      style={{ border: "none" }}
                      title="Valid Gaming License"
                    ></iframe>
                  </Styles.Gaming>

                  <div dangerouslySetInnerHTML={{ __html: fields.copy }} />
                </Styles.CopySection>
              </Theme.Wrap>
            </Styles.Footer>
          </>
        )
      }}
    </Context.Consumer>
  )
}

export default Footer
