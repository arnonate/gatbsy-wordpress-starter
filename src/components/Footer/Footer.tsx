import React from "react"

import { DataProps as LayoutDataProps } from "../Layout/Layout"

type ComponentProps = {
  data: LayoutDataProps
}

const Footer = ({ data }: ComponentProps) => (
  <footer>
    &copy; {new Date().getFullYear()} {data.site.siteMetadata.title}. A
    Gatsby/WP starter by <a href="https://twitter.com/arnonate">arnonate</a>
  </footer>
)

export default Footer
