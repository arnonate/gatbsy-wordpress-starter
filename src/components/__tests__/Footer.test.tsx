import React from "react"
import { render } from "@testing-library/react"

import Footer from "../Footer/Footer"
import { DataProps as LayoutDataProps } from "../Layout/Layout"

const data: LayoutDataProps = {
  logo: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: "img src",
        srcSet: "img srcSet",
        sizes: "img sizes",
      },
    },
  },
  site: {
    siteMetadata: {
      title: "Headless",
      author: "@arnonate",
    },
  },
}

describe("Footer", () => {
  it("renders correctly", () => {
    const component = <Footer data={data} />

    const { container } = render(component)

    expect(container.firstElementChild).toMatchSnapshot()
  })
})
