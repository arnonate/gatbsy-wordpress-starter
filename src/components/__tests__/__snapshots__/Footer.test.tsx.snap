// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <p>
    Â© 
    2020
     
    Headless
    . A Gatsby/WP starter by
     
    <a
      href="https://twitter.com/arnonate"
    >
      .
    </a>
  </p>
  <p>
    <a
      href="https://github.com/arnonate/gatbsy-wordpress-starter"
    >
      View on GitHub
    </a>
  </p>
</footer>
`;
